capstone/scripts/extract_cites.py

```diff
@@ -209,24 +217,31 @@ def resolve_full_cite(eyecite_cite):
             if extra:
                 pin_cites.append(extra)
 
-        # create new cite
-        normalized_forms = first_cite.normalized_forms
-        # NOTE if adding any fields here, also add to cite_key()
-        extracted_cite = ExtractedCitation(
-            **normalized_forms,
-            reporter=first_cite.corrected_reporter(),
-            category=category,
-            cited_by=case,
-            target_case_id=target_case_id,
-            target_cases=target_cases,
-            groups=first_cite.groups or {},
-            metadata=metadata,
-            pin_cites=pin_cites,
-            weight=weight,
-            year=first_cite.year,
-        )
+        opinion_includes_cite = {}
+
+        for cite in filtered_cluster:
+            # If citation is already included in this opinion, skip
+            if cite.opinion_id in opinion_includes_cite:
+                continue
 
-        found_cites.append(extracted_cite)
+            # NOTE if adding any fields here, also add to cite_key()
+            extracted_cite = ExtractedCitation(
+                **normalized_forms,
+                reporter=reporter,
+                category=category,
+                cited_by=case,
+                target_case_id=target_case_id,
+                target_cases=target_cases,
+                groups=groups,
+                metadata=metadata,
+                pin_cites=pin_cites,
+                weight=weight,
```

Comment:
pin_cites and weight are wrong here -- they'll redundantly include all the pin_cites for all opinions.

This also makes me think it would be tricky with the nested structure to frame a query like "get cases that cite this case ordered by how many times they cited to it" ... maybe that gets denormalized somewhere.

Reply:
i understand the issue with pin_cites -- how do you feel about only storing the `pin_cites` value for the first entry? I believe this should still support the search you're thinking about. To separate out the problem, we're looking for a query like this:

```
nested query A -> bool/must -> another nested query B with filters for extracted_citations.field = somevalue
```

were we to extend the function, i would imagine us doing:
1. index another field on the first `extracted_citation` value in the cluster that represents the number of documents in the nested structure (we can use script-based sorting, but its not as efficient.)
2. add another query within the **outer** nested structure, kinda like so:

```
nested query A -> bool/must -> another nested query B with filters for extracted_citations.field = somevalue
                                             \_-> another nested query C that checks where extracted_citations.normalized_cite = {our_identifier} and extracted_citations.citecount > {our_threshold}
```

there is certainly a little more complexity here, though I'm not sure what % is designed by us and what % is inherent to how elasticsearch has approached nesting in general.

can you dig into the issue with weight again? It looks to me like weight is just the size of the cluster, with the assumption being that more frequently cited items are more weighty. I suppose there is a limited denormalization here because they're duplicates, but couldn't we say the same about all the other fields? (reporter, category, cited_by, target_case_id, target_cases, groups, metadata)?

Reply:
changed this logic to:
- extract pin_cite at time of citation.
- extract weight after a `group_by` operation that splits each cluster by opinion_id.